### DISCUSSION TITLE
Behaviour of `monai.transforms.RandCropByLabelClasses` when labels are at borders.

### DISCUSSION THREAD

**Discussion by YerePhy** [Project-MONAI/MONAI#4348] [Q&A] [answered]
Hi, 

What this class exactly do when the labels are in the borders or there is not enough image to extract a patch of a specific size. For example, I want a patch of 3x3 if the following array:

```
[
   [0, 0, 0, 1],
   [0, 0, 0, 0],
   [0, 0, 0, 0]
]
```

Does `monai.transforms.RandCropByLabelClasses` pad the image automatically or should I do it previously with another class?

**Comment by Nic-Ma** [comment, marked as answer]
Hi @YerePhy ,

Thanks for your interest here.
In `RandCropByLabelClasses`, we have the logic to automatically adjust the crop center based on image size:
https://github.com/Project-MONAI/MONAI/blob/dev/monai/transforms/utils.py#L425

Thanks.

**Reply by YerePhy** [reply]
I might need to look at the function with a little bit more detail. However, it seems like it does not pad if the center is non valid, just moves the center to a valid point in the image. Let me look at the function with a little bit more detail, give a good explanation for the users and then mark this as answered.

PD: what about improving docstring giving a more datiled description of the behaviour of `RandCropByLabelClasses`?

**Reply by Nic-Ma** [reply]
OK, thanks for your feedback, I will try to do improve it later.

**Reply by Nic-Ma** [reply]
Submitted PR for it: https://github.com/Project-MONAI/MONAI/pull/4362.

Thanks.

**Reply by YerePhy** [reply]
Your welcome, thanks to you too for being so resolutive!

**Comment by Sembelawi** [comment]
What I am doing wrong  if I get the following value error;

"random crop ratios must match the number of indices of classes, got 3 and 512 "

**Reply by KumoLiu** [reply]
Hi @Sembelawi, the error message shows that your ratios number does not equal the label classes. The number of indices of classes means your label has 152 unique numbers. Please check your data.
https://github.com/Project-MONAI/MONAI/blob/facf17693410d41170edd8e94364b4f341369aea/monai/transforms/utils.py#L619-L622

**Reply by Sembelawi** [reply]
Thank you @KumoLiu
